loader_hub/googledrive/base.py

```diff
@@ -0,0 +1,220 @@
+        """Get file ids present in folder/ file id
+        Args:
+            folder_id: folder id of the folder in google drive.
+            file_id: file id of the file in google drive
+        Returns:
+            metadata: List of metadata of filde ids.
+        """
+        try:
+
+            service = build("drive", "v3", credentials=creds)
+            if folder_id:
+                fileids_meta = []
+                query = "'" + folder_id + "' in parents"
+                results = service.files().list(
+                    q=query, fields="*").execute()
+                items = results.get("files", [])
+                for item in items:
+                    if item["mimeType"] == 'application/vnd.google-apps.folder':
+                        fileids_meta.extend(
+                            self._get_fileids_meta(folder_id=item["id"]))
+                    else:
+                        fileids_meta.append(
+                            (item["id"], item["owners"][0]['displayName'], item["name"], item["createdTime"], item["modifiedTime"]))
+            else:
+                # Get the file details
+                file = service.files().get(fileId=file_id, fields="*").execute()
+
+                # Get metadata of the file
+                fileids_meta = (file["id"], file["owners"][0]['displayName'],
+                                file["name"], file["createdTime"], file["modifiedTime"])
+
+            return fileids_meta
+
+        except Exception as e:
+            logger.error(
+                'An error occurred while getting fileids metadata: {}'.format(e))
+
+    def _download_file(self, fileid: str, filename: str) -> None:
+        """Download the file with fileid and filename
+        Args:
+            fileid: file id of the file in google drive
+            filename: filename with which it will be downloaded
+        Returns:
+            None
+        """
+        try:
+            file = drive.CreateFile({'id': fileid})
+            # download file with filename
+            file.GetContentFile(filename)
+        except Exception as e:
+            logger.error(
+                'An error occurred while downloading file: {}'.format(e))
+
+    def _load_data_fileids_meta(self, fileids_meta: str) -> List[Document]:
+        """Load data from fileids metadata
+        Args:
+            fileids_meta: metadata of fileids in google drive.
+        Returns:
+            Lis[Document]: List of Document of data present in fileids
+        """
+
+        try:
+            with tempfile.TemporaryDirectory() as temp_dir:
+                def get_metadata(filename):
+                    return metadata[filename]
+
+                temp_dir = Path(temp_dir)
+                metadata = {}
+
+                for fileid_meta in fileids_meta:
+                    filename = fileid_meta[2]
+                    filepath = os.path.join(temp_dir, filename)
```

Comment:
Will there be issues here when there are nested files/directories?

Reply:
yes it works.